### PR TITLE
Environment Parameter Page and Variable List with backward compatibility on the VE  and Environment Page

### PR DESCRIPTION
**Pull request by manas-droid**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced the application addition process within environments, including UI improvements.
	- Introduced new UI elements and improved layouts in the variable addition page for better user interaction.
	- Implemented advanced variable management features, including the ability to add variables to all environments and improved variable type handling.

- **Refactor**
	- Updated several methods for better error handling and validation when adding variables.

- **Style**
	- Adjusted UI styles, including button text sizes and grid layouts, for a more intuitive user experience.

### CODE REVIEW

File: Ginger/Ginger/Environments/AddEnvironmentWizardLib/AddNewEnvAppsPage.xaml.cs [hunk 1]
```diff
@@ -89,5 +91,19 @@ private void xAddAppBtn_Click(object sender, RoutedEventArgs e)
                 }
             }
         }
+
+        private void AddApplicationToSolution(object sender, RoutedEventArgs e)
+        {
+            AddApplicationPage applicationPage = new(WorkSpace.Instance.Solution);
+
+            applicationPage.ShowAsWindow();
```
Comment (Maheshkale447) [inline Ginger/Ginger/Environments/AddEnvironmentWizardLib/AddNewEnvAppsPage.xaml.cs:99]: Test if relevant agent is also getting added if not existing

File: Ginger/Ginger/Variables/AddVariablePage.xaml [hunk 2]
```diff
@@ -21,7 +24,41 @@
...
+                        <StackPanel Orientation="Horizontal" DockPanel.Dock="Top">
+                            <StackPanel Orientation="Horizontal">
+                                <Label Style="{StaticResource $LabelStyle}" MinWidth="50" Content="Description:"/>
+                                <TextBox Height="25" Width="250"  Style="{StaticResource $TextBoxStyle}"  Name="variableDescription"/>
+                            </StackPanel>
+
+                            <StackPanel Orientation="Horizontal" Name="ValueStackPanel" Visibility="Collapsed" Margin="5, 0, 0, 0">
+                                <Label Style="{StaticResource $LabelStyle}" MinWidth="50" Content="Initial Value:"/>
+                                 <TextBox  Height="25" Width="250" Style="{StaticResource $TextBoxStyle}"  Name="variableValue"/>
+                                 <TextBlock Text="Value is mandatory" Name="ValueError" DockPanel.Dock="Bottom" Foreground="Red" Visibility="Hidden" Margin="5, 0, 0 , 0"/>
```
Comment (Maheshkale447) [inline Ginger/Ginger/Variables/AddVariablePage.xaml:48]: Add controls for Marking variable as input, output, mandatory and publish

File: Ginger/Ginger/Variables/AddVariablePage.xaml.cs [hunk 3]
```diff
@@ -93,69 +144,179 @@ private void XSharedRepoTabListView_MouseDoubleClick(object sender, System.Windo
...
+            varToAdd.Description = Description;
+            varToAdd.SetInitialValue(Value);
+
+            ObservableList<ProjEnvironment> ProjEnvironments = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>();
+
+            ProjEnvironments.ForEach((projEnv) =>
+            {
+                projEnv.StartDirtyTracking();
+
+                projEnv.Applications.ForEach((envApp) =>
```
Comment (Maheshkale447) [inline Ginger/Ginger/Variables/AddVariablePage.xaml.cs:268]: apply where to select only relevant application

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/Variables/AddVariablePage.xaml (hunk 2 above)
Comment (manas-droid) [reply Ginger/Ginger/Variables/AddVariablePage.xaml:48]: Added the required controls panel

File: Ginger/Ginger/Variables/AddVariablePage.xaml.cs (hunk 3 above)
Comment (manas-droid) [reply Ginger/Ginger/Variables/AddVariablePage.xaml.cs]: Implemented this

**Review by Maheshkale447 — APPROVED** [review]

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**